src/store.rs

```diff
@@ -61,14 +84,167 @@ impl Store {
 
         value += increment;
 
-        self.set(key.to_string(), value.to_string().into());
+        state.set(key.to_string(), value.to_string().into());
 
         Ok(value)
     }
+
+    pub fn remove_expired_keys(&self) -> Option<Instant> {
+        let mut state = self.lock();
+        let now = Instant::now();
+
+        let expired_keys: Vec<(Instant, String)> = state
+            .ttls
+            .iter()
+            .take_while(|(expires_at, _)| expires_at <= &now)
+            .cloned()
+            .collect();
+
+        for (when, key) in expired_keys {
+            state.remove(&key);
+            state.ttls.remove(&(when, key));
+        }
+
+        state.ttls.iter().next().map(|&(expires_at, _)| expires_at)
+    }
 }
 
-impl Default for Store {
-    fn default() -> Self {
-        Self::new()
+type Key = String;
+
+pub struct Value {
+    pub data: Bytes,
+    pub expires_at: Option<Instant>,
+}
+
+pub struct NewValue {
+    pub data: Bytes,
+    pub ttl: Option<Duration>,
+}
+
+impl Value {
+    pub fn new(value: Bytes) -> Value {
+        Value {
+            data: value,
+            expires_at: None,
+        }
+    }
+}
+
+pub struct State {
+    keys: HashMap<Key, Value>,
+    ttls: BTreeSet<(Instant, String)>,
```

Comment:
```suggestion
    ttls: BTreeSet<(Instant, Key)>,
```

[Q] I assume you went with `BTreeSet<(Instant, Key)>` instead of `BTreeMap<Instant, Key>` to allow for multiple keys expiring at the same instant, correct?

Reply:
Correct, the other solution would be to use the map and have a Vec on the value but seems more complicated.

Reply:
Yeah, I think the number of keys with ttl collisions would be extremely low unless a specific application, due to particular business needs, has many keys with the same ttl.
Maybe the most readable and simple solution would be to have a BTreeMap? Not sure.

Reply:
`BTreeSet` is more readable and the implementation is simpler.

`BTreeMap<Instant, Vec<Key>>` would require pushing into the vector for insert and iterating through it on expiration.